sources/libClangSharp/CMakeLists.txt

```diff
@@ -37,12 +31,25 @@ find_package(Clang REQUIRED CONFIG
              PATHS ${LLVM_SEARCH_PATHS}
              NO_DEFAULT_PATH)
 
-if(WIN32)
-  add_library(libClangSharp SHARED ${SOURCES})
-  target_include_directories(libClangSharp PUBLIC ${LLVM_INCLUDE_DIRS})
-  target_link_libraries(libClangSharp ${LIBS})
+add_library(ClangSharp SHARED ${SOURCES})
+
+if (TARGET clang-cpp) # Linux packages install libclang and clang-cpp
+    target_link_libraries(ClangSharp PRIVATE clang-cpp libclang)
 else()
-  add_library(ClangSharp SHARED ${SOURCES})
-  target_include_directories(ClangSharp PUBLIC ${LLVM_INCLUDE_DIRS})
-  target_link_libraries(ClangSharp ${LIBS})
+    target_link_libraries(ClangSharp PRIVATE clangAST clangFrontend libclang)
 endif()
+
+target_include_directories(ClangSharp PRIVATE ${CLANG_INCLUDE_DIRS})
+set_target_properties(ClangSharp PROPERTIES
+    PREFIX lib
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION}
+)
+
+include(GNUInstallDirs)
```

Comment:
What's the behavior of this on Windows and is it going to do the "right" thing where Windows considers `*.dll` a `bin` file but `*.lib` a library file (in contrast to Unix which considers both `*.so` and `*.a` to be library files)?

Reply:
On Windows _BINDIR is "bin", _LIBDIR is "lib", .dll files are `RUNTIME`, .lib files are `ARCHIVE'. CMake documents that `[here](https://cmake.org/cmake/help/latest/command/install.html#targets).